apis/python/CMakeLists.txt

```diff
@@ -0,0 +1,332 @@
+
+# Find Python - Use Development.Module (not Development) for manylinux compatibility
+# Development.Module only requires headers/metadata, not Python_LIBRARIES
+find_package(Python COMPONENTS Interpreter Development.Module REQUIRED)
+
+# Get version from scikit-build-core or setuptools_scm
+# scikit-build-core passes the version via SKBUILD_PROJECT_VERSION
+if(DEFINED SKBUILD_PROJECT_VERSION)
+    set(VERSION_STRING "${SKBUILD_PROJECT_VERSION}")
+    message(STATUS "Package version (from scikit-build-core): ${VERSION_STRING}")
+    
+    # Validate that setuptools_scm worked (version should not be unknown)
+    if(VERSION_STRING MATCHES "0\\.0\\.0\\+unknown")
+        message(FATAL_ERROR "setuptools_scm failed to determine version. Version is '${VERSION_STRING}'. "
+                "This likely means git tags are not available or the repository structure is incorrect.")
+    endif()
+elseif(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/RELEASE-VERSION")
+    # Fallback to RELEASE-VERSION file if it exists (for non-scikit-build builds)
+    file(READ "${CMAKE_CURRENT_SOURCE_DIR}/RELEASE-VERSION" VERSION_STRING)
+    string(STRIP "${VERSION_STRING}" VERSION_STRING)
+    message(STATUS "Package version (from RELEASE-VERSION): ${VERSION_STRING}")
+else()
+    # This should only happen in legacy builds or manual CMake invocations
+    set(VERSION_STRING "0.0.0.dev0")
+    message(WARNING "No version available. If building via pip/cibuildwheel, this indicates a problem.")
+endif()
+
+# Build libtiledbsoma using ExternalProject
+# This will build libtiledbsoma and all its dependencies via vcpkg
+# Check for the sdist-vendored directory first
+set(LIBTDB_SDIST_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/dist_links/libtiledbsoma")
+
+if(EXISTS "${LIBTDB_SDIST_SOURCE_DIR}")
+    # We are in an sdist build. Use the vendored path.
+    message(STATUS "Using vendored libtiledbsoma source from sdist")
+    set(LIBTILEDBSOMA_SOURCE_DIR "${LIBTDB_SDIST_SOURCE_DIR}")
+else()
+    # We are in a dev (Git) build. Use the relative path.
+    message(STATUS "Using libtiledbsoma source from Git repo")
+    set(LIBTILEDBSOMA_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../libtiledbsoma")
+endif()
+set(LIBTILEDBSOMA_BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/libtiledbsoma-build")
+set(LIBTILEDBSOMA_INSTALL_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/install")
+
+# Include ExternalProject
+include(ExternalProject)
+
+# Prepare CMake configure arguments for libtiledbsoma
+set(LIBTILEDBSOMA_CMAKE_ARGS
+    -DCMAKE_BUILD_TYPE=Release
+    -DCMAKE_INSTALL_PREFIX=${LIBTILEDBSOMA_INSTALL_PREFIX}
+    -DOVERRIDE_INSTALL_PREFIX=OFF
+    -DTILEDB_REMOVE_DEPRECATIONS=ON
+    -DSUPERBUILD=ON
+)
+
+# Forward compilers if explicitly set
+if(CMAKE_C_COMPILER)
+    list(APPEND LIBTILEDBSOMA_CMAKE_ARGS
+        -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+    )
+endif()
+
+if(CMAKE_CXX_COMPILER)
+    list(APPEND LIBTILEDBSOMA_CMAKE_ARGS
+        -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+    )
+endif()
```

Comment:
These variables will always have a value, since the `program()` command will search for compilers. And even if they didn't have a value, you don't need an `if`; `-DCMAKE_C_COMPILER=` is valid.